src/main/java/zd/zero/waifu/motivator/plugin/settings/WaifuMotivatorSettingsPage.java

```diff
@@ -174,6 +212,73 @@ private void setFieldsFromState() {
         this.frustrationProbabilitySlider.setValue( this.state.getProbabilityOfFrustration() );
         this.enableExitCodeNotifications.setSelected( this.state.isExitCodeNotificationEnabled() );
         this.enableExitCodeSound.setSelected( this.state.isExitCodeSoundEnabled() );
+        initializeExitCodes();
     }
 
+    private void initializeExitCodes() {
+        int preExistingRows = exitCodeListModel.getRowCount();
+        if ( preExistingRows > 0 ) {
+            IntStream.range( 0, preExistingRows )
+                .forEach( idx -> exitCodeListModel.removeRow( 0 ) );
+        }
+        Arrays.stream( this.state.getAllowedExitCodes()
+            .split( WaifuMotivatorState.DEFAULT_DELIMITER ) )
+            .map( Integer::parseInt )
+            .forEach( exitCodeListModel::addRow );
+        exitCodesChanged = false;
+    }
+
+    private boolean exitCodesChanged = false;
+
+    private void createUIComponents() {
+        exitCodeListModel = new ListTableModel<Integer>() {
```

Comment:
I took the code from here
https://github.com/JetBrains/intellij-community/blob/33e818537a52f6337a268c9db803eef5bc78f549/platform/platform-impl/src/com/intellij/ide/plugins/PluginHostsConfigurable.java#L108

![image](https://user-images.githubusercontent.com/15972415/93003791-52be4980-f507-11ea-8d13-b38309217cd7.png)
:point_up: I saw that and was like "I want to use that"